library/Core/src/Juvix/Core/IR/Evaluator.hs

```diff
@@ -203,22 +233,26 @@ toLambda' π' ty' pats rhs = do
       IR.Term' (OnlyExts.T z) primTy primVal
     lam x = IR.Lam' x ()
 
+-- | Extract variable from a pattern definition.
 singleVar :: Alternative f => IR.Pattern' ext primTy primVal -> f IR.PatternVar
 singleVar (IR.PVar' p _) = pure p
 singleVar _ = empty
 
+-- | Discard annotations from a `Term'`, but keep the extensions.
```

Comment:
So what is the purpose of these functions?

Reply:
github seems to have eaten this response ugh

the `toLambda[R]` functions don't assume that the global context has the same extension as the term being inlined into, which means there needs to be a way to translate from one to the other. the main purpose of the `toOnlyExts{T,E}` functions is to do this translation, which requires some restrictions:

- the input ext (the one in the globals) must have no extension constructors, since it wouldn't know how to translate them. this is the `NoExtensions` constraint
- the output ext can't have any non-`()` annotations, since it can't necessarily produce them. this is the `OnlyExts.T` wrapper

this is obviously not the only way to solve this. in hindsight it would probably have been better to just restrict the global and term exts to be the same.

anyway, the new descriptions you wrote for these functions sound good to me, so i've approved the PR